src/shell/mod.rs

```diff
@@ -0,0 +1,88 @@
+//! Unified wayland and xdg shell surface handling
+use wayland_client::protocol::{wl_output, wl_seat, wl_surface};
+use wayland_client::Proxy;
+
+use wayland_protocols::xdg_shell::client::xdg_toplevel;
+pub use wayland_protocols::xdg_shell::client::xdg_toplevel::State;
+
+use Shell;
+
+mod wl;
+mod xdg;
+mod zxdg;
+
+/// Possible events generated by a window that you need to handle
+#[derive(Clone, Debug)]
+pub enum Event {
+    /// The state of your window has been changed
+    Configure {
+        /// Optional new size for your *inner* surface
```

Comment:
This is no longer true. The events generated by the shell surfaces are the new size for the *whole* surface, including all decorations and everything. It is the `Window` abstraction that processes it to subtract the decoration size.

Also considering that the `Refresh` does not exist for a bare shell surface, I think it'd be better to create an `Event` enum specifically for the shell surface exposition, rather than re-use the `Window` one.